docs/trestle-spec.md

```diff
@@ -0,0 +1,525 @@
+# Trestle Specifications (v0.0.1)
+
+## Table of Contents
+
+- [Purpose](<#purpose>)
+- [Users](<#users>)
+- [Scope](<#scope>)
+- [Trestle commands](<#trestle-commands>)
+  - [Draft commands](<#draft-commands>)
+- [Future work](<#future-work>)
+  - [Deploy commands](<#deploy-commands>)
+  - [Monitor commands](<#monitor-commands>)
+  - [Reporting commands](<#reporting-commands>)
+
+## Purpose
+
+This document contains detail specifications of the Trestle commands.
+
+Trestle offers various commands to simplify operations at different steps in compliance management and reporting.
+
+Trestle assumes all security and compliance specifications and requirements are expressed in OSCAL format.
+
+## Users
+
+Trestle aims at compliance engineers who are familiar with various software development tools such as Git, CI/CD and command line tools.
+
+Users of Trestle are also expected to be comfortable with editing OSCAL files in YAML/JSON/XML format.
+
+## Scope
+
+The scope of this document is to describe the purpose and expected behaviour of various trestle commands for manipulating OSCAL documents ONLY. This will not be all of trestle. Workflow commands will be subsequent / expanded on this.
+
+## Trestle Commands
+
+### Draft Commands
+
+For the draft phase of compliance engineering, trestle provides the following commands to facilitate various draft related operations.
+
+
+#### `trestle init`
+
+This command will create a trestle project in the current directory with necessary directory structure and trestle artefacts. For example, if we run `trestle init` in a directory, it will create a directory structure like below for different artefacts as well as initiaze `git` in it:
```

Comment:
Okay I'm not sure we want to tie so closely to git. My comment was more towards 'lets remove references to git if they are not required'. I do not think we are formally exploiting git yet so I think stripping references would be safer for now?

Reply:
Makes sense. References removed.